File: server/models/Message.js
const mongoose = require('mongoose');

const MessageSchema = new mongoose.Schema({
  chatId: {
    type: mongoose.Schema.Types.ObjectId,
    ref: 'Chat',
    required: true,
  },
  userId: {
    type: mongoose.Schema.Types.ObjectId,
    ref: 'User',
    required: true,
  },
  role: {
    type: String,
    enum: ['user', 'assistant', 'system'],
    required: true,
  },
  content: {
    type: String,
    required: true,
  },
  model: {
    type: String,
    default: 'system', // e.g., 'gemini-2.5-flash'
  },
  latencyMs: {
    type: Number,
  },
  emotion: {
    type: String,
    enum: ['idle', 'thinking', 'speaking', 'listening', 'happy', 'concerned', 'encouraging', 'calm'],
    default: 'idle',
  },
  sourceDocs: [{
    docId: mongoose.Schema.Types.ObjectId,
    snippet: String,
  }],
  actions: [{
    type: { type: String }, // e.g., 'goal_created'
    payload: Object,
  }],
  timestamp: {
    type: Date,
    default: Date.now,
  }
});

module.exports = mongoose.model('Message', MessageSchema);

File: flutter_client/lib/features/chat/models/message.dart
enum MessageRole { user, ai, system }

class ChatMessage {
  final String id;
  final String content;
  final MessageRole role;
  final DateTime timestamp;

  ChatMessage({
    required this.id,
    required this.content,
    required this.role,
    DateTime? timestamp,
  }) : timestamp = timestamp ?? DateTime.now();
  
  ChatMessage copyWith({String? content}) {
    return ChatMessage(
      id: id,
      content: content ?? this.content,
      role: role,
      timestamp: timestamp,
    );
  }
}
